src/Observers/ItemObserver.php

```diff
@@ -59,14 +59,18 @@ protected function cleanModel(MenuItem $item)
         if (request()->has('link_type')) {
             // Prevent multi-types : keep only the value on submitted type
             $attributes = MenuItem::linkTypesAttributes();
-            if (array_key_exists(request()->get('link_type'), $attributes)) {
-                $attrToClean = collect($attributes)->forget(request()->get('link_type'));
-                foreach ($attrToClean as $attr) {
-                    $item->{$attr} = null;
-                }
+            if(request()->get('link_type') === MenuItem::TYPE_EMPTY) {
```

Comment:
You have to clean other fields too. Example : if previous type was "internal" : you have to clean old value.